integration/query_array_test.go

```diff
@@ -137,3 +140,96 @@ func TestQueryArraySize(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryArrayDotNotation(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{{"_id", "array-double"}, {"value", bson.A{float64(1)}}},
+		bson.D{{"_id", "array-strings"}, {"value", bson.A{"foo", "bar", "baz"}}},
+		bson.D{
+			{"_id", "document-array-field"},
+			{"value", bson.D{{"array", bson.A{int32(0), nil}}}},
+		},
+		bson.D{
+			{"_id", "document-document-field"},
+			{"value", bson.D{{"document", bson.D{{"foo", nil}}}}},
+		},
+	})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		expectedIDs []any
+		err         *mongo.CommandError
+	}{
+		"PositionDouble": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "double"}}}},
+			expectedIDs: []any{"array-double"},
+		},
+		"PositionString": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "string"}}}},
+			expectedIDs: []any{"array-strings"},
+		},
+		"PositionNull": {
```

Comment:
If it refers to `PositionRegex` and it was expected to put here nil, not `null` string value, then here is a typo. 
If it's not a type I suggest renaming the test.